HW2/P5/physics.pyx

```diff
@@ -121,11 +150,14 @@ cpdef update(FLOAT[:, ::1] XY,
         # SUBPROBLEM 1: parallelize this loop over 4 threads (with static
         #    scheduling).
         # SUBPROBLEM 2: update the grid values.
-        for i in range(count):
+
+        # for i in range(count):
+        for i in prange(count, schedule='static', chunksize=count/4, num_threads=1):
             for dim in range(2):
                 XY[i, dim] += V[i, dim] * t
 
 
+
 def preallocate_locks(num_locks):
     cdef omp_lock_t *locks = get_N_locks(num_locks)
     assert 0 != <uintptr_t> <void *> locks, "could not allocate locks"
```

Comment:
I could not figure out anything special to say about this problem. Clearly styled.